### PR TITLE
Mkdir

### PR DESCRIPTION
**Pull request by hm-lnkd**
finished mkdir

### CODE REVIEW

File: src/mkdir.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    for (auto i = input.find("r") + 1; i < input.size(); i++) {
+        auto character = input.at(i);
+        if (isspace(character)) {
+            if (not foldername.empty()) {
+                sub1(foldername);
+                foldername.clear();
+            }
+        } else {
+            foldername += character;
+        }
```
Comment (Nicholas-Baron) [inline src/mkdir.cpp]:
```suggestion
        if(auto character = input.at(i); not isspace(character)) {
            foldername += character;
        } else if (not foldername.empty()) {
            sub1(foldername);
            foldername.clear();
        }
```

File: src/mkdir.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#include "commands.h"
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <cctype>
+#include <iostream>
+void sub1(const std::string & input);
```
Comment (Nicholas-Baron) [inline src/mkdir.cpp]: This should be a lambda inside `mkdir` or be a `static` function. Either way, forward declaration is less maintainable than simple declaration.

**Review by Nicholas-Baron — CHANGES_REQUESTED** [review]
A simple test for `mkdir` would be nice.

File: src/mkdir.cpp (hunk 2 above)
Comment (hm-lnkd) [reply src/mkdir.cpp]: how would I specifically avoid forward declaration

File: src/mkdir.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#include "commands.h"
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <cctype>
+
+// sub-function made to differ the mkdir() functions
+void sub1(const std::string & input) {
```
Comment (Nicholas-Baron) [inline src/mkdir.cpp]:
```suggestion
static void sub1(const std::string & input) {
```

File: src/mkdir.cpp [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    if (mkdir(input.c_str(), S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH) == -1)
+        perror("mkdir");
+}
+
+// command for mkdir
+void seashell::mkdir(const std::string & input) {
+    std::string foldername;
+    for (auto i = input.find("r") + 1; i < input.size(); i++) {
```
Comment (Nicholas-Baron) [inline src/mkdir.cpp]:
```suggestion
    for (auto i = input.find('r') + 1; i < input.size(); i++) {
```

**Review by Nicholas-Baron — CHANGES_REQUESTED** [review]
These two changes and a `make format` should be good for a merge.

**Comment by hm-lnkd** [comment]
accepted changes, make format gave no differences

**Review by Nicholas-Baron — APPROVED** [review]

**Pull request merged**